### PR TITLE
Resx string editor quality-of-life improvements

### PR DESCRIPTION
**Pull request by wjk**
This PR does two things:

1. Fixes the bug described in #2881.
2. Changes the `ResourceStringTable` to automatically change the size of its rows to show the complete cell value immediately upon committing edits to a cell; previously you had to manually click out of and back into an affected cell to have the row resize. This makes editing long strings in resx files much less annoying.

If there is a way to ensure that the text box used to edit the value itself similarly resizes to fit its text, I'm all ears, but I don't know if that is possible. Thanks!

### CODE REVIEW

File: src/Microsoft.VisualStudio.Editors/ResourceEditor/ResourceStringTable.vb [hunk 1]
```diff
@@ -1869,6 +1870,14 @@ Namespace Microsoft.VisualStudio.Editors.ResourceEditor
             End Sub
         End Class
 
+        Protected Overrides Function ProcessDataGridViewKey(e As KeyEventArgs) As Boolean
+            If e.KeyCode = Keys.Space AndAlso e.Shift Then
+                Return False
+            Else
+                Return MyBase.ProcessDataGridViewKey(e)
```
Comment (Pilchie) [inline src/Microsoft.VisualStudio.Editors/ResourceEditor/ResourceStringTable.vb]: What does the DataGridView normally treat Shift+Space as?  Will this change break expectations for something people expect from DGV in other applications?

Comment (wjk) [reply src/Microsoft.VisualStudio.Editors/ResourceEditor/ResourceStringTable.vb]: As far as I can tell, Shift+Space selects either the entire row, or the entire column, depending on the value of `DataGridView.SelectionMode`. In practice, this highlights the entire row, without losing the edit focus. This has no practical benefit (to me, at least), and it also eats the space input, which is extremely annoying when typing text with the spacebar held down (to type all-caps text, or to type an ampersand between words, for example). I have never heard of anyone actually using this "feature" anywhere, and in fact I'd reckon that other DataGridView users have become just as confused by the same quirk that the resx editor suffers from.

Comment (davkean) [reply src/Microsoft.VisualStudio.Editors/ResourceEditor/ResourceStringTable.vb]: I think I'd prefer this to be limited to the only when editing the content of the cells, there's no reason to eat CTRL+Space when not in editing mode. This lets you easily select and delete a cell when you moving the selection around for a single cell.

File: src/Microsoft.VisualStudio.Editors/ResourceEditor/ResourceStringTable.vb [hunk 2]
```diff
@@ -87,6 +87,7 @@ Namespace Microsoft.VisualStudio.Editors.ResourceEditor
             AllowUserToDeleteRows = False
             EditMode = DataGridViewEditMode.EditOnKeystrokeOrF2
             AutoSizeColumnsMode = DataGridViewAutoSizeColumnsMode.Fill
+            AutoSizeRowsMode = DataGridViewAutoSizeRowsMode.AllCells
```
Comment (davkean) [inline src/Microsoft.VisualStudio.Editors/ResourceEditor/ResourceStringTable.vb]: To call out the behavior change for reviewers; this means that rows will automatically be sized to fit content - and you can no longer use mouse to resize heights. I'm okay with this.

File: src/Microsoft.VisualStudio.Editors/ResourceEditor/ResourceStringTable.vb (hunk 1 above)
Comment (davkean) [reply src/Microsoft.VisualStudio.Editors/ResourceEditor/ResourceStringTable.vb]: It looks like DataGridView.IsCurrentCellInEditMode indicates that.

**Comment by davkean** [comment]
This looks great. I just made one change - I pushed it down to the base class so that the Settings editor also gets the new behavior.

**Comment by davkean** [comment]
Merged this in, thanks again for the contribution!

**Pull request merged**